stat/mds/mds.go

```diff
@@ -14,21 +14,19 @@ import (
 // TorgersonScaling converts a dissimilarity matrix to a matrix containing
 // Euclidean coordinates. TorgersonScaling places the coordinates in dst and
 // returns it and the number of positive Eigenvalues if successful.
-// If the scaling is not successful, dst is returned, but will not be a valid scaling.
-// When the scaling is successful, mds will be resized to k columns wide.
+// If the scaling is not successful, dst will be empty upon return.
+// When the scaling is successful, dst will be resized to k columns wide.
 // Eigenvalues will be copied into eigdst and returned as eig if it is provided.
 //
-// If dst is nil, a new mat.Dense is allocated. If dst is not empty,
-// the dimensions of dst and dis must match otherwise TorgersonScaling will panic.
-// The dis matrix must be square or TorgersonScaling will panic.
-func TorgersonScaling(dst *mat.Dense, eigdst []float64, dis mat.Symmetric) (k int, mds *mat.Dense, eig []float64) {
+// TorgersonScaling will panic if dst is not empty.
+func TorgersonScaling(dst *mat.Dense, eigdst []float64, dis mat.Symmetric) (k int, eig []float64) {
```

Comment:
Rename this to `TorgersonScalingTo`?

Reply:
If you want ... I personally don't think it's better. In `mat`, things are confusing because input matrices can be sources or destinations. There it's useful to have `To` in the signature, for example to distinguish between `SetFromU` and `UTo`. Or perhaps more importantly, many of the operations modify the receiver, so it's important to distinguish those that don't.

On the flip side, if you look at somewhere like `distmv`, I feel like if you see the signature of `Normal.CovarianceMatrix(dst *mat.Dense)`, I don't think there's any confusion that the behavior stores the covariance matrix in `dst`. The `To` then just adds clutter to the names.

Reply:
Fair enough.